core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -385,6 +369,40 @@ sealed abstract class MultiSigSigner
 
 object MultiSigSigner extends MultiSigSigner
 
+/** Used to sign a [[org.bitcoins.core.protocol.script.P2SHScriptPubKey]] */
+sealed abstract class P2SHSigner extends BitcoinSigner[P2SHSpendingInfo] {
+  override def sign(
```

Comment:
I don't think that this method currently handles `P2SH(P2WPKH)` and `P2SH(P2WSH)` correctly (see `TxBuilder.signAndAddInput` for more details)